### DISCUSSION TITLE
REAPER (libswell, for menus and windows)

### DISCUSSION THREAD

**Discussion by rgm89git** [catppuccin/catppuccin#2640] [Port Requests]
### Will you make the port?

- [X] Yes, I'm willing to make the port

### How can the suggested port be themed?

First, copying the color theme file:
`cp /usr/lib/REAPER/libSwell.colortheme ~/.config/REAPER/libSwell-user.colortheme`

This file will include the configuration to change the colors.
I can't find any documentation about this file, but this [**theme apply helper**](https://github.com/iliaspoulakis/Reaper-Tools/blob/master/LibSwell%20Colorize/Colorize%20unthemeable%20areas%20to%20match%20active%20theme.lua) i found can help.

![image](https://github.com/user-attachments/assets/3bff3dc0-704b-4929-be7b-2b4b102f61aa)
![image](https://github.com/user-attachments/assets/5c57282a-8e5e-4244-bd80-7c4c92e0378b)
![image](https://github.com/user-attachments/assets/f47a7c63-3031-488b-a599-66e6bb6e54d4)

### Submission Guidelines

- [X] I have read and followed Catppuccin's submission guidelines.

**Comment by rgm89git** [comment]
Here's the mapping that script uses, for reference:

**MENU:**
**Menubar** -> 'menubar_bg'
**Menubar Text** -> 'menubar_text'

**WINDOW:**
**Main Background** -> '_3dface', 'edit_bg_disabled'
**Main Text** -> 'checkbox_text', 'edit_text', 'label_text', 'tab_text', 'group_text'
**Main Text Disabled** -> 'checkbox_text_disabled', 'edit_text_disabled', 'label_text_disabled'
**Main Shadow** -> 'group_shadow'
**Main Highlight** -> 'group_hilight'

**MENU:**
**Menu Background** -> 'menu_bg', 'treeview_bg', 'listview_bg', 'info_bk', 'trackbar_track', 'trackbar_mark', 'scrollbar_bg', '_3dshadow', '_3dhilight', '_3ddkshadow', 'tab_shadow', 'tab_hilight'
**Menu Text** -> 'menu_text', 'treeview_text', 'listview_text', 'listview_hdr_text', 'info_text', 'focusrect', 'edit_cursor', 'combo_text'
**Menu Text Disabled** -> 'menu_text_disabled', 'menubar_text_disabled', 'combo_text_disabled'
**Menu Highlight** -> 'menu_hilight'
**Menu Shadow** -> 'menu_shadow'

**MENU SELECT:**
**Menu Selected Background** -> 'menubar_bg_sel', 'menu_bg_sel', 'combo_arrow_press', 'checkbox_inter', 'edit_bg_sel', 'trackbar_knob', 'progress', 'listview_bg_sel', 'treeview_bg_sel', 'focus_hilight'
**Menu Selected Text** -> 'menubar_text_sel', 'menu_text_sel', 'edit_text_sel', 'listview_text_sel', 'treeview_text_sel'

**MENU EXTRAS:**
**Menu Header** -> 'listview_hdr_bg', 'menu_scroll', 'scrollbar'
**Menu Header Shadow** -> 'listview_hdr_shadow', 'listview_hdr_hilight', 'listview_grid'
**Menu Arrow** -> 'checkbox_fg', 'combo_arrow', 'treeview_arrow', 'menu_scroll_arrow', 'listview_hdr_arrow', 'menu_submenu_arrow'
**Menu Scroll** -> 'scrollbar_fg'

**EDIT / COMBO / CHECKBOX**
**Edit (Edit Background)** -> 'edit_bg'
**Edit Border** -> 'edit_shadow', 'edit_hilight', 'button_shadow', 'button_hilight'
**Combo** -> 'combo_bg', 'combo_bg2', 'combo_shadow', 'combo_hilight'
**Checkbox** -> 'checkbox_bg'

**BUTTON:**
**Button Background** -> 'button_bg'
**Button Text** -> 'button_text'
**Button Text Disabled** -> 'button_text_disabled'
**Button Border** -> 'button_shadow', 'button_hilight'

**Comment by ItsProfessional** [comment]
For what it's worth, there's also this script: https://forum-amz.cockos.com/showthread.php?t=241630
Also, swell is open source: https://github.com/justinfrankel/WDL/blob/main/WDL/swell/swell.cpp

**Comment by Ecstaticoder** [comment]
Any updates on this?
